dot/state/block.go

```diff
@@ -336,19 +336,20 @@ func (bs *BlockState) AddBlockWithArrivalTime(block *types.Block, arrivalTime ui
 	hash := block.Header.Hash()
 
 	// set best block key if this is the highest block we've seen
-	// TODO: update to use canonical path
-	if block.Header.Number.Cmp(bs.highestBlockHeader.Number) == 1 {
+	if hash == bs.BestBlockHash() {
 		err = bs.setBestBlockHashKey(hash)
 		if err != nil {
 			return err
 		}
 	}
 
-	// TODO: only set number->hash mapping for our canonical chain, otherwise, this messes up BlockResponses
-	// store number to hash
-	err = bs.db.Put(headerHashKey(block.Header.Number.Uint64()), hash.ToBytes())
-	if err != nil {
-		return err
+	// only set number->hash mapping for our current chain
+	var ok bool
```

Comment:
this can also be declared on the line below since it is not used outside the condition block whether you keep `ok` or change it to something more descriptive

Reply:
it's defined outside because otherwise there's a lint error saying definition of err is being shadowed